modules/serverless-common/pages/service-mesh/common-service-mesh-setup.adoc

```diff
@@ -1,28 +1,82 @@
-= Setup Eventing with {SMProductName}
+= Setup {serverlessproductname} with {SMProductName}
 :compat-mode!:
 // Metadata:
-:description: Setup Eventing with {SMProductName}
+:description: Setup {serverlessproductname} with {SMProductName}
 
 // TODO
 
+.Assumptions and limitations
+
+This page describes the integration of {serverlessproductname} with {smproductname}. In this setup, it is assumed that:
+
+* All Knative internal components, as well as Knative Services, are part of the {smproductshortname} and have istio-sidecars injected. This means, that strict mTLS is enforced within the whole mesh. All requests to Knative Services require an mTLS connection  (the client must send its certificate), except calls coming from OpenShift Routing.
+* {serverlessproductname} with Kourier disabled can only target *one* istio service mesh. Multiple meshes can be present in the cluster, but {serverlessproductname} will only be available on one of them.
+* The mesh that {serverlessproductname} is part of has to be distinct and ideally only for {serverlessproductname} workloads, as additional configuration like `Gateways` might interfere with the automated configuration by {serverlessproductname}. Other namespaces can be part of this Mesh, but must not configure the istio custom resource `Gateway` that could overlap with {serverlessproductname} `knative-local-gateway` and `knative-ingress-gateway`.
+* {smproductname} only allows one `Gateway` to claim a wildcard host binding (`hosts: *`) on the same port (`port: 443`). So the `knative-ingress-gateway` and `knative-local-gateway` have to be unique in the Mesh. If another Mesh is already binding this configuration, a separate Mesh has to be created for {serverlessproductname} workloads.
+* Cluster external Knative Services are expected to be called via OpenShift Ingress using OpenShift Routes. It is not supported to access istio directly, e.g. by exposing the `istio-ingressgateway` using a Service with type `NodePort` or `LoadBalancer`.
+* Changing the target `ServiceMeshMemberRole`, that {serverlessproductname} is part of (meaning moving {serverlessproductname} to another mesh) is not supported. The only way to change the targeted Service Mesh is to uninstall and reinstall {serverlessproductname}.
+
```

Comment:
This list is very technical and goes into a lot of details that the reader might not know. We should give here concrete examples how (a) the can check whether the assumption are valid for the cluster (i.e. concrete command line commands how to do so) and (b) what to do to if the assumption do not hold and how to fix this.

This is not only helpful for our users but also for supporting/debugging a setup if we have concrete checks. I'm not sure if this is fully possible, but we should at least try this. Ideally we could also offer a script (like the helm script) that does these preflight checks automatically. Then we could also avoid to describe these where implementation specific details at this place and just say: "Run this script". If this preflight script fails it should also print out instructions how fix any issue.

Reply:
I like the idea. Let me think about it a bit.

But even if we have a script, we still need to "declare" those. All of those came from real customer cases, so we need to have it in the docs to say what do we support and what not. Also people could change stuff later, even if the script works fine at installation time.

Reply:
I agree, so having a staged approach with two sections (a simplified one with a script and one with the background information) but be helpful.

Regardless what, I still believe we that for the details descriptions we should give hints how to check the preconditions, even when it needs some manual inspections with `oc`.

Reply:
I added some pre-installation verification steps. PTAL.
But only some of the assumptions can be checked directly. Other things are a bit more complex to identify. So in the end, it is still up the what the user already has as existing setup. To limit discussions that we already have in support cases, lets keep the other bullet points in there.

Reply:
Thanks a lot, I think this extended checklist is perfectly fine for this documentation update.